src/k8s/pkg/client/kubernetes/certificate_signing_request.go

```diff
@@ -0,0 +1,54 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+
+	certificatesv1 "k8s.io/api/certificates/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WatchCertificateSigningRequest watches a CertificateSigningRequest with the
+// given name and calls a verify function on each event.
+// WatchCertificateSigningRequest will continue watching the CSR until the
+// verify function returns true or an non-retriable error occurs.
+// WatchCertificateSigningRequest will return true and a wrapped error if the
+// error is retriable.
+// WatchCertificateSigningRequest will return false and a wrapped error if the
+// error is not retriable.
```

Comment:
This does not help if I read this out of context. It would be easier if we included the retry loop in case the watcher fails, then this godoc can say: we keep repeating as long as verify returns false or "retriable" errors. Because right now the responsibility of retrying the watcher is split across two very far places in the code.